security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt

```diff
@@ -60,84 +63,65 @@ class AesEncryptionManagerTest {
         given(keyStorage.store(MockUtils.any())).willAnswer { it.arguments.first() }
-        val setupObserver = TestObserver<Boolean>()
         preferences.remove(PREF_KEY_PASSWORD_CHECKSUM)
 
         // Setup with "test"
-        manager.setupPassword("test".toByteArray()).subscribe(setupObserver)
+        assertTrue(manager.setupPassword("test".toByteArray()))
         then(keyStorage).should().store(MockUtils.any())
-        setupObserver.assertNoErrors().assertValue(true)
 
         // Check that it is unlocked
-        val unlockedObserver = TestObserver<Boolean>()
-        manager.unlocked().subscribe(unlockedObserver)
-        unlockedObserver.assertNoErrors().assertValue(true)
+        assertTrue(manager.unlocked())
 
         // Check that data can be en- and decrypted
         val encryptedData = manager.encrypt("Hello World".toByteArray())
         assertEquals("Hello World", String(manager.decrypt(encryptedData)))
 
         // Check that password cannot be changed if one is already set
-        val invalidChangeObserver = TestObserver<Boolean>()
-        manager.setupPassword("test2".toByteArray()).subscribe(invalidChangeObserver)
-        invalidChangeObserver.assertNoErrors().assertValue(false)
+        assertFalse(manager.setupPassword("test2".toByteArray()))
 
         // Check that password can be changed with old password
-        val validChangeObserver = TestObserver<Boolean>()
-        manager.setupPassword("test2".toByteArray(), "test".toByteArray()).subscribe(validChangeObserver)
-        validChangeObserver.assertNoErrors().assertValue(true)
+        assertTrue(manager.setupPassword("test2".toByteArray(), "test".toByteArray()))
 
         // Check that device can be locked
         manager.lock()
-        val lockedObserver = TestObserver<Boolean>()
-        manager.unlocked().subscribe(lockedObserver)
-        lockedObserver.assertNoErrors().assertValue(false)
+        assertFalse(manager.unlocked())
 
         // Check that device cannot be unlocked with wrong password
-        val wrongPasswordObserver = TestObserver<Boolean>()
-        manager.unlockWithPassword(("invalid".toByteArray())).subscribe(wrongPasswordObserver)
-        wrongPasswordObserver.assertNoErrors().assertValue(false)
+        assertFalse(manager.unlockWithPassword(("invalid".toByteArray())))
 
-        // Check that device can be unlocked with new password
-        val unlockObserver = TestObserver<Boolean>()
-        manager.unlockWithPassword(("test2".toByteArray())).subscribe(unlockObserver)
-        unlockObserver.assertNoErrors().assertValue(true)
+        // Check that device can be unlocked with old password
+        assertTrue(manager.unlockWithPassword(("test2".toByteArray())))
 
         // Check that data encrypted with old password can still be decrypted
         assertEquals("Hello World", String(manager.decrypt(encryptedData)))
     }
 
     @Test
-    fun fingerprintFlow() {
+    fun fingerprintFlow() = runBlocking {
```

Comment:
Maybe we should be using `runBlockingTest`? I think the only difference is that it ensures the serialised execution of tests.

Reply:
`runBlockingTest` doesn't wait if there is a `delay`. So IMHO we should prefer it for tests.